src/types/middleware.ts

```diff
@@ -0,0 +1,28 @@
+import { EffectScope } from 'vue'
+import { MaybePromise } from '@/types/utilities'
+
+type Route = {
+  name: string,
+  params: Record<string, unknown>,
+  query: unknown,
+  hash: string,
+}
+
+interface RouteState {
+
+}
+
+interface RouteRejection {
+  NotFound: unknown,
+}
+
+type RouteReject = (type?: keyof RouteRejection) => void
+
+type MiddleWareExtras = {
```

Comment:
nit, I like middleware as 1 word

```suggestion
type MiddlewareExtras = {
```